js/repl/CodeMirror.js

```diff
@@ -39,22 +40,27 @@ export default class ReactCodeMirror extends React.Component {
   };
 
   _codeMirror: any;
-  _textAreaRef: HTMLTextAreaElement;
+  _container: HTMLDivElement;
+
+  shouldComponentUpdate() {
+    return false;
+  }
 
   componentDidMount() {
-    this._codeMirror = CodeMirror.fromTextArea(this._textAreaRef, {
+    this._codeMirror = CodeMirror(this._container, {
+      value: this.props.value || "",
+      placeholder: this.props.placeholder,
+      autofocus: this.props.autoFocus,
       ...DEFAULT_CODE_MIRROR_OPTIONS,
       ...this.props.options,
     });
     this._codeMirror.on("change", this._onChange);
-    this._codeMirror.setValue(this.props.value || "");
   }
 
   componentWillUnmount() {
-    // is there a lighter-weight way to remove the cm instance?
-    if (this._codeMirror) {
-      this._codeMirror.toTextArea();
-    }
+    const container = this._container;
+    container.removeChild(container.children[0]);
+    this._codeMirror = null;
```

Comment:
I guess we didn't actually need to call `toTextArea()` ? 😄

Reply:
As the source is not a textarea anymore, it doesn't make sense to convert to to textarea back. I stole this from somewhere, but cannot recall from where.